testing/test_assertion.py

```diff
@@ -2198,3 +2198,38 @@ def test_vvv():
         ]
     )
     result.stdout.no_fnmatch_line(expected_non_vvv_arg_line)
+
+
+import re
+
+
+def test_dict_extra_items_preserve_insertion_order(pytester):
+    pytester.makepyfile(
+        test_order="""
+        def test_order():
+            a = {
+                "first": 1,
```

Comment:
Please let's use a dict with more keys for this test, only 2 seems few to catch any regressions. I suggest using simply `"a"`, `"b"`, etc, inserted in non-alphabetical order.